torax/transport_model/tglf_based_transport_model.py

```diff
@@ -0,0 +1,206 @@
+    """Shared parameters for TGLF-based models."""
+
+    def make_provider(
+        self, torax_mesh: geometry.Grid1D | None = None
+    ) -> "RuntimeParamsProvider":
+        return RuntimeParamsProvider(**self.get_provider_kwargs(torax_mesh))
+
+
+@chex.dataclass(frozen=True)
+class DynamicRuntimeParams(quasilinear_transport_model.DynamicRuntimeParams):
+    """Shared parameters for TGLF-based models."""
+
+    pass
+
+
+@chex.dataclass
+class RuntimeParamsProvider(runtime_params_lib.RuntimeParamsProvider):
+    """Provides a RuntimeParams to use during time t of the sim."""
+
+    runtime_params_config: RuntimeParams
+
+    def build_dynamic_params(self, t: chex.Numeric) -> DynamicRuntimeParams:
+        return DynamicRuntimeParams(**self.get_dynamic_params_kwargs(t))
+
+
+@chex.dataclass(frozen=True)
+class TGLFInputs(quasilinear_transport_model.QuasilinearInputs):
+    r"""Dimensionless inputs to the TGLF model.
+
+    See https://gafusion.github.io/doc/tglf/tglf_table.html for definitions.
+    """
+
+    # Ti/Te
+    Ti_over_Te: chex.Array
+    # dRmaj/dr
+    dRmaj: chex.Array
+    # q
+    q: chex.Array
+    # r/q dq/dr
+    s_hat: chex.Array
+    # nu_ee (see note in prepare_tglf_inputs)
+    nu_ee: chex.Array
+    # Elongation, kappa
+    kappa: chex.Array
+    # Shear in elongation, r/kappa dkappa/dr
+    kappa_shear: chex.Array
+    # Triangularity, delta
+    delta: chex.Array
+    # Shear in triangularity, r ddelta/dr
+    delta_shear: chex.Array
+    # Electron pressure defined w.r.t B_unit
+    beta_e: chex.Array
+    # Effective charge
+    Zeff: chex.Array
+
+
+class TGLFBasedTransportModel(quasilinear_transport_model.QuasilinearTransportModel):
+    """Base class for TGLF-based transport models."""
+
+    def _prepare_tglf_inputs(
+        Zeff_face: chex.Array,
+        nref: chex.Numeric,
+        q_correction_factor: chex.Numeric,
+        transport: DynamicRuntimeParams,
+        geo: geometry.Geometry,
+        core_profiles: state.CoreProfiles,
+    ) -> TGLFInputs:
+        # Note: TGLF uses geo.rmid = (Rmax - Rmin)/2 as the radial coordinate
+        # This means all gradients are calculated w.r.t. rmid
+
+        ## Shorthand for commonly used variables
+        Te = core_profiles.temp_el
+        Ti = core_profiles.temp_ion
+        ne = core_profiles.ne
+
+        ## Reference velocity and length, used for normalisation
+        # https://gafusion.github.io/doc/cgyro/outputs.html#output-normalization
+        vref = (Te.face_value() / (core_profiles.Ai * CONSTANTS.mp)) ** 0.5
+        lref = geo.Rmin[-1]  # Minor radius at LCFS
+
+        ## Temperature gradients, At = -1/T * dT/drho
+        # Note: RLTS = a_minor / T * dT/dr = 1 / T * dT/drho
+        # https://gafusion.github.io/doc/tglf/tglf_table.html#id2
+        Ti_over_Te = Ti.face_value() / Te.face_value()
+        Ate = -1 / Te.face_value() * Te.face_grad(geo.rmid)
+        Ati = -1 / Ti.face_value() * Ti.face_grad(geo.rmid)
```

Comment:
Note that you can use the new `quasilinear_transport_model.NormalizedLogarithmicGradients.from_profiles`  for these. See

https://github.com/google-deepmind/torax/blob/4812e48c15d2ace8eadcd50cd43c043a7a9f904b/torax/transport_model/qualikiz_based_transport_model.py#L138

1. I don't recommend on using the names Ate, Ati, etc., which are QuaLiKiz jargon. In any case Normalized logarithmic gradients contains general names
2. Note that Ati, Ate, (in QuaLiKiz jargon) should already include the normalization length, which is aminor for TGLF. NormalizedLogarithmicGradients takes the reference length as input